src/long.ts

```diff
@@ -291,6 +290,67 @@ export class Long extends BSONValue {
     return result;
   }
 
+  /**
+   * Returns a Long representation of the given string, written using the specified radix.
+   * Throws an error if any of the following conditions are true:
+   * - the string contains invalid characters for the given radix
+   * - the string contains whitespace
+   * - the value the string represents is too large or too small to be a Long
+   * @param str - The textual representation of the Long
+   * @param unsigned - Whether unsigned or not, defaults to signed
+   * @param radix - The radix in which the text is written (2-36), defaults to 10
+   * @returns The corresponding Long value
+   */
+  static fromStringStrict(str: string, unsigned?: boolean, radix?: number): Long {
+    if (str === 'NaN' || str === 'Infinity' || str === '+Infinity' || str === '-Infinity')
+      return Long.ZERO;
```

Comment:
This would be a silent coercion from unsupported string values, I thought this was undesirable?

Reply:
Bookkeeping: Current changes only coerce  `+/-Infinity` and `NaN` cases to `Long.Zero` in `Long.fromString` when the radix does not support their respective letters.